src/ActionsSimplificationModal.js

```diff
@@ -0,0 +1,186 @@
+// @flow
+import React from 'react';
+import { injectIntl } from 'react-intl';
+import type {IntlShape} from 'react-intl';
+import ModalWithFooter from './ModalWithFooter';
+
+import ActionsMenuItem from './ActionsMenuItem';
+import ProgramSequence from './ProgramSequence';
+
+import type {ActionToggleRegister, CommandName} from './types';
+import ModalHeader from './ModalHeader';
+import {extend} from './Utils';
+import {ReactComponent as SimplificationIcon} from './svg/Simplification.svg'
+
+import './ActionsSimplificationModal.scss';
+
+type ActionsSimplificationModalProps = {
+    intl: IntlShape,
+    onCancel: () => void,
+    onConfirm: (allowedActions: ActionToggleRegister) => void,
+    // TODO: Flesh this definition out.
+    menuItems: {},
+    programSequence: ProgramSequence,
+    allowedActions: ActionToggleRegister,
+    show: boolean
+};
+
+type ActionsSimplificationModalState = {
+    allowedActions: ActionToggleRegister
+}
+
+class ActionsSimplificationModal extends React.Component<ActionsSimplificationModalProps, ActionsSimplificationModalState> {
+    static defaultProps = {
+        menuItems: {
+            forward1: {
+                isAllowed: true,
+                labelKey: "Command.forward1"
+            },
+            forward2: {
+                isAllowed: true,
+                labelKey: "Command.forward2"
+            },
+            forward3: {
+                isAllowed: true,
+                labelKey: "Command.forward3"
+            },
+            backward1: {
+                isAllowed: true,
+                labelKey: "Command.backward1"
+            },
+            backward2: {
+                isAllowed: true,
+                labelKey: "Command.backward2"
+            },
+            backward3: {
+                isAllowed: true,
+                labelKey: "Command.backward3"
+            },
+            left45: {
+                isAllowed: true,
+                labelKey: "Command.left45"
+            },
+            left90: {
+                isAllowed: true,
+                labelKey: "Command.left90"
+            },
+            left180: {
+                isAllowed: true,
+                labelKey: "Command.left180"
+            },
+            right45: {
+                isAllowed: true,
+                labelKey: "Command.right45"
+            },
+            right90: {
+                isAllowed: true,
+                labelKey: "Command.right90"
+            },
+            right180: {
+                isAllowed: true,
+                labelKey: "Command.right180"
+            }
+        }
+    }
+
+    constructor(props: ActionsSimplificationModalProps) {
+        super(props);
+        this.state = {
+            allowedActions: this.props.allowedActions
+        }
+    }
+
+    render() {
+        const cancelButtonProperties = {
+            label: this.props.intl.formatMessage({ id: 'ActionsSimplificationModal.cancel'},),
```

Comment:
We could clean up this syntax a little, by removing , inside of formatMessage method.

Reply:
Good catch.

Reply:
There's another same syntax issue for the saveButtonProperties.

Reply:
Fixed.